tools/template.go

```diff
@@ -0,0 +1,140 @@
+package tools
```

Comment:
This was renamed from `test_utils.go` to align it a bit more with the `thinking` package